modules/cloud-run-v2/service.tf

```diff
@@ -0,0 +1,232 @@
+        }
+        dynamic "startup_probe" {
+          for_each = containers.value.startup_probe == null ? [] : [""]
+          content {
+            initial_delay_seconds = containers.value.startup_probe.initial_delay_seconds
+            timeout_seconds       = containers.value.startup_probe.timeout_seconds
+            period_seconds        = containers.value.startup_probe.period_seconds
+            failure_threshold     = containers.value.startup_probe.failure_threshold
+            dynamic "http_get" {
+              for_each = containers.value.startup_probe.http_get == null ? [] : [""]
+              content {
+                path = containers.value.startup_probe.http_get.path
+                dynamic "http_headers" {
+                  for_each = coalesce(containers.value.startup_probe.http_get.http_headers, tomap({}))
+                  content {
+                    name  = http_headers.key
+                    value = http_headers.value
+                  }
+                }
+              }
+            }
+            dynamic "tcp_socket" {
+              for_each = containers.value.startup_probe.tcp_socket == null ? [] : [""]
+              content {
+                port = ontainers.value.startup_probe.tcp_socket.port
+              }
+            }
+            dynamic "grpc" {
+              for_each = containers.value.startup_probe.grpc == null ? [] : [""]
+              content {
+                port    = containers.value.startup_probe.grpc.port
+                service = containers.value.startup_probe.grpc.service
+              }
+            }
+          }
+        }
+      }
+    }
+    dynamic "volumes" {
+      for_each = var.volumes
+      content {
+        name = volumes.key
+        dynamic "secret" {
+          for_each = volumes.value.secret == null ? [] : [""]
+          content {
+            secret       = volumes.value.secret.name
+            default_mode = volumes.value.secret.default_mode
+            dynamic "items" {
+              for_each = volumes.value.secret.path == null ? [] : [""]
+              content {
+                path    = volumes.value.secret.path
+                version = volumes.value.secret.version
+                mode    = volumes.value.secret.mode
+              }
+            }
+          }
+        }
+        dynamic "cloud_sql_instance" {
+          for_each = length(coalesce(volumes.value.cloud_sql_instances, [])) == 0 ? [] : [""]
+          content {
+            instances = volumes.value.cloud_sql_instances
+          }
+        }
+        dynamic "empty_dir" {
+          for_each = volumes.value.empty_dir_size == null ? [] : [""]
+          content {
+            medium     = "MEMORY"
+            size_limit = volumes.value.empty_dir_size
+          }
+        }
+      }
+    }
+  }
+
+  lifecycle {
+    ignore_changes = [
+      template.0.annotations["run.googleapis.com/operation-id"],
+    ]
+  }
+}
+
+resource "google_cloud_run_service_iam_binding" "binding" {
```

Comment:
Not sure of the differences but maybe worth to change this resource to ' google_cloud_run_v2_service_iam_binding'